core/blockchain.go

```diff
@@ -2132,17 +2132,57 @@ func (bc *BlockChain) NextCXReceiptsProofUnpentCheckpoint(currentNum uint64, sha
 	return newCheckpoint
 }
 
-// UpdateCXReceiptsProofUnspentAndCheckpoint will update the checkpoint and clean unspent receipts upto checkpoint
-func (bc *BlockChain) UpdateCXReceiptsProofUnspentAndCheckpoint(currentNum uint64, shardID uint32) {
+// cleanCXReceiptsCheckpoints will update the checkpoint and clean spent receipts upto checkpoint
+func (bc *BlockChain) cleanCXReceiptsCheckpoints(shardID uint32, currentNum uint64) {
 	lastCheckpoint, err := rawdb.ReadCXReceiptsProofUnspentCheckpoint(bc.db, shardID)
 	if err != nil {
-		utils.Logger().Warn().Msg("[UpdateCXReceiptsProofUnspentAndCheckpoint] Canot get lastCheckpoint")
+		utils.Logger().Warn().Msg("[cleanCXReceiptsCheckpoints] Canot get lastCheckpoint")
 	}
-	newCheckpoint := bc.NextCXReceiptsProofUnpentCheckpoint(currentNum, shardID)
+	newCheckpoint := bc.NextCXReceiptsCheckpoint(currentNum, shardID)
 	if lastCheckpoint == newCheckpoint {
 		return
 	}
+	utils.Logger().Debug().Uint64("lastCheckpoint", lastCheckpoint).Uint64("newCheckpont", newCheckpoint).Msg("[CleanCXReceiptsCheckpoints]")
 	for num := lastCheckpoint; num < newCheckpoint; num++ {
 		rawdb.DeleteCXReceiptsProofUnspent(bc.db, shardID, num)
 	}
 }
+
+// MarkCXReceiptsSpent mark the incomingReceipts in a block as spent
+// avoid double spent
+func (bc *BlockChain) MarkCXReceiptsSpent(block *types.Block) {
+	for _, cxp := range block.IncomingReceipts() {
+		rawdb.WriteCXReceiptsProofUnspent(bc.db, cxp)
```

Comment:
This func should be named WriteCXReceiptsProofSpent, right?

Reply:
My original thought is mark cxreceiptsproof as spent